release-22.2: sql: cleanup independent error messages for missing users

otan opened the pull request:
Backport 1/1 commits from #94677.

/cc @cockroachdb/release

Release justification: risk free cleanup of error messaging

---

The missing user error message has lots of different forms in the codebase. This unifies to the one, whilst also fixing some missing error codes.

Epic: none

Release note (bug fix): Previously, certain GRANT or REVOKE commands on a user which does not exist would error with the correct PG code. This is now fixed.

Release note (sql change): Error messages for missing users sometimes had different forms. This is now unified as `role/user "user" does not exist`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94735)
<!-- Reviewable:end -->

rafiss reviewed (changes requested):
the pkg/sql/scanner/token_names_test.go file should not be committed

otan committed:
sql: cleanup independent error messages for missing users

The missing user error message has lots of different forms in the
codebase. This unifies to the one, whilst also fixing some missing error codes.

Epic: none

Release note (bug fix): Previously, certain GRANT or REVOKE commands on
a user which does not exist would error with the correct PG code. This
is now fixed.

Release note (sql change): Error messages for missing users sometimes
had different forms. This is now unified as `role/user "user" does not
exist`.

otan reviewed (commented):
oops
